src/main/java/com/github/pgasync/impl/netty/NettyPgProtocolStream.java

```diff
@@ -119,6 +124,21 @@ public NettyPgProtocolStream(EventLoopGroup group, SocketAddress address, boolea
         }).lift(throwErrorResponsesOnComplete());
     }
 
+    private void scheduleSubscriptionCleaner() {
+        Runnable action = () ->
+                Optional.ofNullable(subscribers.peek())
+                        .filter(Subscriber::isUnsubscribed)
+                        .ifPresent(s -> {
+                            ctx.fireExceptionCaught(new IllegalStateException());
+                            ctx.close();
+                        });
```

Comment:
this exception will break correctly working communication which is taking more time than 1s.

Reply:
If it's a timeout due to long running query than it maybe worth to close connection.